### PR TITLE
added Disable .sort for .toSorted lint rule

### PR DESCRIPTION
**Pull request by pnaik1**
Closes: https://issues.redhat.com/browse/RHOAIENG-2414

## Description
added lint rule to disable .sort, use .toSorted instead

## How Has This Been Tested?

1. Check the Resource page whether the vertical tabs are sorted accordingly
2. Check in Resourec page the doc card are sorted accordingly
3. Check the Serving runtimes Page has the list sorted 

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/components/table/useTableColumnSort.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ const useTableColumnSort = <T>(
         return data;
       }
 
-      return [...data].sort((a, b) => {
+      return [...data].toSorted((a, b) => {
```
Comment (christianvogt) [inline frontend/src/components/table/useTableColumnSort.ts:92]:
```suggestion
      return data.toSorted((a, b) => {
```

File: frontend/src/app/AppNotificationDrawer.tsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ interface AppNotificationDrawerProps {
 
 const AppNotificationDrawer: React.FC<AppNotificationDrawerProps> = ({ onClose }) => {
   const stateNotifications: AppNotification[] = useAppSelector((state) => state.notifications);
-  const notifications = [...stateNotifications].sort(
+  const notifications = [...stateNotifications].toSorted(
```
Comment (christianvogt) [inline frontend/src/app/AppNotificationDrawer.tsx:28]:
```suggestion
  const notifications = stateNotifications.toSorted(
```

File: frontend/src/app/AppLauncher.tsx [hunk 3]
```diff
@@ -110,7 +110,7 @@ const AppLauncher: React.FC = () => {
       return acc;
     }, getODHApplications());
 
-    sections.sort((a, b) => sectionSortValue(a) - sectionSortValue(b));
+    sections.toSorted((a, b) => sectionSortValue(a) - sectionSortValue(b));
 
     return sections;
```
Comment (christianvogt) [inline frontend/src/app/AppLauncher.tsx:115]: No longer returns the sorted esections.

**Comment by pnaik1** [comment]
Updated the PR accordingly

**Pull request merged**